src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1531,6 +1531,7 @@ private static bool TryGetExplicitImplementationAssociatedPropertyOrEvent(Method
                     interfaceMethod.TypeParameters,
                     interfaceMethod.Parameters,
                     interfaceMethod.RefKind,
+                    interfaceMethod.IsInitOnly,
```

Comment:
>interfaceMethod.IsInitOnly, [](start = 20, length = 27)

It is not obvious what implication this might have. If it is redundant because we would require modifiers to match, the why do we even need it? If it is not redundant, then could we fail to match in some scenarios where runtime would?

Reply:
Maybe I don't understand the question. 
The OHI checks that were added look for `IsInitOnly` property, but here's were substituting a symbol for another, so we need to maintain the value of that property. 
Maybe we could implement `SignatureOnlyMethodSymbol.IsInitOnly` by looking at modreqs (should be in `ReturnTypeWithAnnotations`). Is that what you're suggesting?

---
In reply to: [411000780](https://github.com/dotnet/roslyn/pull/43275#discussion_r411000780) [](ancestors = 411000780)

Reply:
> Maybe I don't understand the question. 

I think we should get more clarity why this is needed. Is this redundant piece of information or not. What scenarios will not work without this and why.

---
In reply to: [411523071](https://github.com/dotnet/roslyn/pull/43275#discussion_r411523071) [](ancestors = 411523071,411000780)

Reply:
If you look at the call to `IsInterfaceMemberImplementation` a couple of lines below, you will see that custom modifiers are sometimes ignored (when `implementingTypeIsFromSomeCompilation` is true). So it was updated to check for `IsInitOnly` property.
But here's we're constructing a replacement symbol for comparison, and lose `IsInitOnly` in the process (`SignatureOnlyMethodSymbol` would always have `IsInitOnly`=false unless we do something).

There are two ways of giving this replacement symbol the appropriate `IsInitOnly` value:
- we can give it directly (as I did here)
- we can reconstruct it from other information available (such as `ReturnTypeWithAnnotations`, which has the custom modifiers)

I suppose that a third option would be to use a comparison ignoring modifiers *except* the IsInitOnly modifier, but that seems even more complicated...

Below is a scenario that misbehaves without this change (produces no warning if `SignatureOnlyMethodSymbol.IsInitOnly` doesn't return proper value):

``` csharp
public interface I
{
    public string Property { get; init; }
}
public class CWithoutInit : I // 1
{
    public string Property { get; set; }
}
```

---
In reply to: [411533357](https://github.com/dotnet/roslyn/pull/43275#discussion_r411533357) [](ancestors = 411533357,411523071,411000780)